client/verta/tests/test_model_registry/test_model_version.py

```diff
@@ -622,6 +681,56 @@ def test_arbitrary_object(self, model_version):
 
         self._assert_no_deployment_artifacts(model_version)
 
+    def test_download_arbitrary_directory(self, model_version, dir_and_files, strs, in_tempdir):
+        """Model that was originally a dir is unpacked on download."""
+        dirpath, _ = dir_and_files
+        download_path = strs[0]
+
+        model_version.log_model(dirpath)
+        model_version.download_model(download_path)
+
+        # contents match
+        utils.assert_dirs_match(dirpath, download_path)
+
+    def test_from_run_download_arbitrary_directory(
+        self,
+        experiment_run,
+        registered_model,
+        dir_and_files,
+        strs,
+        in_tempdir,
+    ):
+        """Dir model logged to run is unpacked by model ver."""
+        dirpath, _ = dir_and_files
+        download_path = strs[0]
+
+        experiment_run.log_model(dirpath)
+        model_version = registered_model.create_version_from_run(
+            run_id=experiment_run.id,
+            name="From Run {}".format(experiment_run.id),
+        )
+        model_version.download_model(download_path)
+
+        utils.assert_dirs_match(dirpath, download_path)
```

Comment:
This is the only _new_ test. The rest were copied nearly verbatim from experiment run tests.